common/src/main/java/com/wynntils/wc/utils/WynnItemMatchers.java

```diff
@@ -29,6 +33,29 @@ public static boolean isHealingPotion(ItemStack stack) {
                                 .contains(ChatFormatting.RED + "Potion of Healing"));
     }
 
+    public static boolean isCraftedHealingPotion(ItemStack stack) {
+        System.out.println(stack.getHoverName());
+
+        if (stack.isEmpty() || stack.getItem() != Items.POTION) return false;
+
+        boolean isCraftedPotion = false;
+        boolean hasHealEffect = false;
+        ListTag lore = ItemUtils.getLoreTagElseEmpty(stack);
+        for (Tag tag : lore) {
+            String unformattedLoreLine = ComponentUtils.getUnformatted(tag.getAsString());
+
+            if (unformattedLoreLine == null) continue;
+
+            if (unformattedLoreLine.equals("Crafted Potion")) {
+                isCraftedPotion = true;
+            } else if (unformattedLoreLine.startsWith("- Heal:")) {
+                hasHealEffect = true;
+            }
```

Comment:
You can cancel the check if there’s another effect that isn’t healing

Reply:
Is that really faster and/or required? Some non-healing pots don't have an effect, only in charge count for example. I think it's just better to not do any indirect decisions.